src/PaymentGateways/PayPalCommerce/AjaxRequestHandler.php

```diff
@@ -104,21 +113,133 @@ public function onGetPartnerUrlAjaxRequestHandler() {
 	 * @since 2.8.0
 	 */
 	public function removePayPalAccount() {
-		$this->sendErrorOnAjaxRequestIfUserDoesNotHasPermission();
+		$this->validateAdminRequest();
 
 		give( MerchantDetail::class )->delete();
 
 		wp_send_json_success();
 	}
 
 	/**
-	 * Send error if user does not has capability to manage GiveWP settings.
+	 * Create order.
+	 *
+	 * @todo: handle payment create error on frontend.
 	 *
 	 * @since 2.8.0
 	 */
-	private function sendErrorOnAjaxRequestIfUserDoesNotHasPermission() {
+	public function createOrder() {
+		$this->validateFrontendRequest();
+
+		/* @var PayPalHttpClient $client */
+		$client = give( PayPalClient::class )->getHttpClient();
+		/* @var MerchantDetail $merchant */
+		$merchant = give( MerchantDetail::class );
```

Comment:
Great time to use automatic dependency injection for the class. If you look at this same class in the webhooks branch you'll see how I'm doing it. Copy it over so we don't get merge conflicts.

Reply:
You already updated code in your webhook pr. I will fix that after merging your pr.